src/bio/bio_xstream.c

```diff
@@ -73,6 +76,85 @@ struct bio_nvme_data {
 static struct bio_nvme_data nvme_glb;
 uint64_t vmd_led_period;
 
+static int64_t
+hexstr_to_int(const char *src)
+{
+	uint64_t val;
+
+	val = strtoul(src, NULL, 16);
+	if (errno == ERANGE) {
+		D_ERROR("%s hex string out of range\n", src);
+		return -1;
+	}
+
+	D_INFO("hex2dec val: %lu\n", val);
+
+	return (int64_t)val;
+}
+
+static int64_t
+traddr_to_bus_id(char *src)
+{
+	char		*tok;
+	int		 i;
+
+	for (i = 0; i < 2; i++) {
+		tok = strtok(src, ":");
+		if (tok == NULL) {
+			D_ERROR("pci address field %d empty\n", i);
+			return -1;
+		}
+		D_INFO("pci address field %d: %s\n", i, tok);
+	}
+
+	return hexstr_to_int(tok);
+}
+
+static bool
+hotplug_filter_fn(const struct spdk_pci_addr *addr)
+{
+	char		str[128];
+	int64_t		low;
+	int64_t		high;
+	int64_t		bus_id;
+
+	if (spdk_pci_addr_fmt(str, sizeof(str), addr) != 0) {
+		D_ERROR("unable to format pci address\n");
+		return false;
+	}
+
+	if ((!g_hotplug_bus_id_low) || (!g_hotplug_bus_id_high))
+		return false;
+
+	low = hexstr_to_int(g_hotplug_bus_id_low);
+	if (low < 0) {
+		D_ERROR("low bus id %s invalid\n", g_hotplug_bus_id_low);
+		return false;
+	}
+
+	high = hexstr_to_int(g_hotplug_bus_id_high);
+	if (high < 0) {
+		D_ERROR("high bus id %s invalid\n", g_hotplug_bus_id_high);
+		return false;
+	}
+
+	bus_id = traddr_to_bus_id(str);
+	if (bus_id < 0) {
+		D_ERROR("input transport address %s invalid\n", str);
+		return false;
+	}
+
+	D_INFO("hotplug: bus id %lX (filter %lX-%lX)\n", bus_id, low, high);
+	if ((bus_id < low) || (bus_id > high)) {
```

Comment:
Do we expect user to set filter for valid bus ID range or excluded bus ID range? I think setting valid (allowed) bus ID looks more straightforward?